Client/Frontend/Shields/ShareShieldsController.swift

```diff
@@ -0,0 +1,280 @@
+        var icon: UIImage {
+            switch self {
+                case .twitter:
+                    return  #imageLiteral(resourceName: "share-twitter")
+                case .facebook:
+                    return  #imageLiteral(resourceName: "share-facebook")
+                default:
+                    return  #imageLiteral(resourceName: "share-mail")
+            }
+        }
+    }
+    
+    // MARK: UX
+    
+    private struct UX {
+        static let defaultOffsetInset: CGFloat = 19
+        static let iconSize: CGFloat = 24
+    }
+    
+    // MARK: Properties
+    
+    private let shareType: ShareType
+    
+    private let contentView = UIView().then {
+        $0.backgroundColor = .clear
+    }
+    
+    private let titleLabel = UILabel().then {
+        $0.textAlignment = .center
+        $0.font = .systemFont(ofSize: 19)
+        $0.numberOfLines = 0
+        $0.setContentHuggingPriority(.required, for: .horizontal)
+    }
+    
+    private let titleImageView = UIImageView().then {
+        $0.contentMode = .scaleAspectFit
+        $0.tintColor = .darkGray
+        $0.setContentCompressionResistancePriority(.required, for: .horizontal)
+    }
+    
+    // MARK: Lifecycle
+    
+    init(frame: CGRect = .zero, type: ShareType) {
+        shareType = type
+        
+        super.init(frame: frame)
+
+        doLayout()
+        setContent()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: Internal
+    
+    private func doLayout() {
+        addSubview(contentView)
+        
+        if shareType == .default {
+            contentView.addSubview(titleLabel)
+        } else {
+            contentView.addSubviews([titleImageView, titleLabel])
+        }
+        
+        contentView.snp.makeConstraints {
+            $0.edges.equalToSuperview().inset(UX.defaultOffsetInset)
+        }
+        
+        if shareType != .default {
+            titleImageView.snp.makeConstraints {
+                $0.width.equalTo(UX.iconSize)
+                $0.height.equalTo(UX.iconSize)
+                $0.trailing.equalTo(titleLabel.snp.leading).offset(-UX.defaultOffsetInset)
+                $0.top.equalTo(contentView.snp.top)
+                $0.bottom.equalTo(titleLabel.snp.bottom)
+            }
+        }
+        
+        let labelOffset = shareType == .default ? 0 : UX.iconSize / 2
+            
+        titleLabel.snp.makeConstraints {
+            $0.top.equalToSuperview()
+            $0.centerX.equalToSuperview().offset(labelOffset)
+            $0.bottom.equalToSuperview()
+        }
+        
+        layer.cornerRadius = 6
+        layer.masksToBounds = true
+    }
+    
+    private func setContent() {
+        titleLabel.text = shareType.title
+        titleImageView.image = shareType.icon
+    }
+    
+    func applyTheme(_ theme: Theme) {
+        backgroundColor = theme.isDark ? UIColor(rgb: 0x303443) : Colors.neutral000
```

Comment:
Is this colour `0x303443` custom and only used in this view? Is it possible to add it to the `Colors` struct if it isn't used just here alone?

Reply:
Yeah this is a color only used once, thats why I didnt add it